dev/design-decisions-readme.md

```diff
@@ -40,3 +40,55 @@ Of course, there are some caveats to this approach, based on the above discussio
 * For test purposes, some applications may need to allow Java system properties to override environment variable values. This allows configuration settings to be easily passed in during tests.
 * Care must be taken to ensure that any environment variable that includes a password, API key, etc. include the word "secret" in its name. The application must then filter any environment variables dumps in error reports, diagnostic pages, etc. to mask such variables' values, to prevent accidental disclosure.
 * Security practices must account for the fact that any attacker that can run commands or code on the system as the application user will be able to gain access to that application's secrets.
+
+## Performance Tests
+
+Benchmarks are hard to get right, and getting them wrong can lead to bad decisions: committing "performance improvements" that, in reality, are performance-neutral, or even performance-negative. The main problem is that program execution times are noisy, which means that benchmark runs must be viewed in a statistical context. The results of a single run can't be trusted, even when it's on the same hardware. In fact, from some preliminary research, it seems that around 30 runs of a given test may be needed to generate a result that one can be confident in. The following paper is strongly recommended reading: [The Speedup-Test: A Statistical Methodology for Program Speedup Analysis and Computation](https://hal.inria.fr/hal-00764454/document/).
+
+The execution environments used across benchmark runs is also a significant concern: results from different environments aren't really comparable. In particular, moving from a single-system environment (e.g. a developer's workstation) to a multi-system environment (e.g. the Blue Button AWS environments, which have at least three separate machines) can **wildly** change a program's execution profile: CPU processing time may be dominant in one environment, while network latencies might be dominant in another. Again: bad test design can lead to performance-negative decisions and outcomes.
+
+To address these concerns, our benchmarks take the following approach:
+
+* Run the benchmarks on production-like clones in AWS. Each benchmark iteration will stand up its own set environment clone and run on it.
```

Comment:
Are we not able to re-purpose existing CloudFormation scripts used to stand up dev/test/prod to provision a benchmark environment?